### ISSUE TITLE
change colour of sort icon on flat table header 

### ISSUE THREAD

**Issue by niamhmadden24** [Sage/carbon#6523]
### Desired behaviour

Sufficient colour contrast on sort header icon. 

### Current behaviour

The colour of the sort icon on the flat table, (see docs: https://carbon.sage.com/?path=/docs/flat-table--with-sorting-headers#with-sorting-headers) doesn't meet colour contrast requirements for a11y. 

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by tomdavies73** [comment]
@harpalsingh @ljemmo seems to be a contrast issue with one of our icons here on Flat Table. Could you advise on an alternative token to use here? Or will we need to have this token updated? Cheers

<img width="742" alt="Screenshot 2024-01-16 at 14 39 04" src="https://github.com/Sage/carbon/assets/114918852/9a1d160a-43f8-4e9f-9eaa-55de519e84a5">

**Comment by ljemmo** [comment]
@tomdavies73 i believe it should be using {colors.action.minor.yang.100}

**Comment by sianford** [comment]
FE-6342
